internal/liqonet/tunnel-operator/reflected_endpointslice_controller.go

```diff
@@ -0,0 +1,283 @@
+		klog.Infof("Endpointslice %q under deletion: skipping iptables rules update", nsName)
+		return ctrl.Result{}, nil
+	}
+
+	//Check if endpoint(s) are no more part of the endpointslice
+	value, loaded := r.endpointslicesInfo.Load(nsName)
+	if loaded {
+		oldEndpointsInfo := value.(map[string]liqoiptables.EndpointInfo)
+		for oldEndpoint, oldEndpointInfo := range oldEndpointsInfo {
+			if _, ok := endpointsInfo[oldEndpoint]; !ok {
+				oldEndpointInfo.Deleting = true
+				endpointsInfo[oldEndpoint] = oldEndpointInfo
+			}
+		}
+	}
+
+	// Check if there aren't new information: in this case it's not necessary ensure iptables rules
+	if len(endpointsInfo) == 0 {
+		// Endpoints fiels of Endpoinslice yet empty: skip creation of iptables rules and return no error
+		klog.Infof("Endpoints of endpointslice %q not yet set: skipping iptables rules update", nsName)
+		return ctrl.Result{}, nil
+	}
+
+	// Store endpointslicesInfo object
+	r.endpointslicesInfo.Store(nsName, endpointsInfo)
+
+	// Ensure iptables rules
+	klog.Infof("Ensuring updated iptables rules")
+	if err := r.gatewayNetns.Do(r.ensureIptablesRules); err != nil {
+		klog.Errorf("Error while ensuring iptables rules: %w", err)
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *reflectedEndpointsliceController) ensureIptablesRules(netns ns.NetNS) error {
+	return r.EnsureRulesForEndpointslicesReflected(r.endpointslicesInfo, r.IPSetHandler)
+}
+
+func (r *reflectedEndpointsliceController) endpointsliceEnqueuer(_ context.Context, obj client.Object) []ctrl.Request {
+	gvks, _, err := r.Scheme.ObjectKinds(obj)
+	// Should never happen, but if it happens we panic.
+	utilruntime.Must(err)
+
+	// If gvk is found we log.
+	if len(gvks) != 0 {
+		klog.Infof("handling resource %q of type %q", klog.KObj(obj), gvks[0].String())
+	}
+
+	endpointslices := discoveryv1.EndpointSliceList{}
+	if err := r.List(context.Background(), &endpointslices, client.InNamespace(obj.GetNamespace())); err != nil {
+		klog.Error(err)
+		return []ctrl.Request{}
+	}
+
+	if len(endpointslices.Items) == 0 {
+		klog.Infof("no endpointslice found for resource %q", klog.KObj(obj))
+		return []ctrl.Request{}
+	}
+
+	requests := []ctrl.Request{}
+	for _, endpointslice := range endpointslices.Items {
+		klog.V(4).Infof("enqueuing endpointslice %q", endpointslice.Name)
+		requests = append(requests, ctrl.Request{NamespacedName: types.NamespacedName{Name: endpointslice.Name, Namespace: obj.GetNamespace()}})
+	}
+
+	return requests
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *reflectedEndpointsliceController) SetupWithManager(mgr ctrl.Manager) error {
+	// endpointslicePredicate selects those endpointslices matching the provided label
+	endpointslicePredicate, err := predicate.LabelSelectorPredicate(metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			discoveryv1.LabelManagedBy: "endpointslice-controller.k8s.io",
+		},
+	})
+	if err != nil {
+		klog.Error(err)
```

Comment:
You don't need to log it, just create a new error with fmt.Errorf and return it.